ext/flash/lib.rs

```diff
@@ -253,9 +253,8 @@ async fn op_flash_write_resource(
           .write_all(b"Transfer-Encoding: chunked\r\n\r\n")
           .await?;
         loop {
-          let vec = vec![0u8; 64 * 1024]; // 64KB
-          let buf = ZeroCopyBuf::new_temp(vec);
-          let (nread, buf) = resource.clone().read_return(buf).await?;
+          let buf = ZeroCopyBuf::new_temp(vec![0u8; 64 * 1024]); // 64KB
+          let (nread, buf) = resource.clone().read_owned(buf).await?;
```

Comment:
this could use `read` if the buffer was outside `loop`?

Reply:
It can actually even use `read_owned` if the buffer is outside the loop. See the latest commit :)

Reply:
I realized that essentially you will always call `read_owned`. Only for `write` does the non-owned version have an actual usecase. PR incoming shortly to demonstrate this.